src/components/commands-panel.tsx

```diff
@@ -0,0 +1,50 @@
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import React from 'react';
+import { LoopForm } from './loop-form';
+import { v4 as uuidv4 } from 'uuid';
+
+export function requestLoop(
+  loopRequestPub: RomiCore.Publisher<RomiCore.Loop> | null,
+  fleetName: string,
+  numLoops: number,
+  startLocationPoint: string,
+  endLocationPoint: string,
+) {
+  loopRequestPub?.publish({
+    finish_name: endLocationPoint,
+    num_loops: numLoops,
+    robot_type: fleetName,
+    start_name: startLocationPoint,
+    task_id: uuidv4(),
+  });
+}
+
+export interface CommandsPanelProps {
+  fleets: readonly RomiCore.FleetState[];
+  transport?: Readonly<RomiCore.Transport>;
+}
+
+export default function CommandsPanel(props: CommandsPanelProps): React.ReactElement {
+  const { fleets, transport } = props;
+
+  const allFleets = fleets.flatMap(fleet => fleet.name);
+  const loopRequestPub = React.useMemo(
+    () => (transport ? transport.createPublisher(RomiCore.loopRequests) : null),
+    [transport],
+  );
+
+  const handleRequestLoop = (
+    fleetName: string,
+    numLoops: number,
+    startLocationPoint: string,
+    endLocationPoint: string,
+  ) => {
+    requestLoop(loopRequestPub, fleetName, numLoops, startLocationPoint, endLocationPoint);
+  };
+
+  return (
+    <React.Fragment>
+      <LoopForm requestLoop={handleRequestLoop} fleets={allFleets} />
+    </React.Fragment>
```

Comment:
Hey @koonpeng would be good to have also an accordion component like this?
![image](https://user-images.githubusercontent.com/11761240/87581662-51a7a200-c6a7-11ea-8db5-0ec62c25a0c6.png)
We will need to add the Delivery Request here too and I think we will add more commands in the future too.

Reply:
So if I understand correctly, each request type will have its own accordion component, right? should this be made in this PR or created in another PR when delivery request is added in?

Reply:
I thought about this before, that would be nice especially as we add more commands.

Reply:
> So if I understand correctly, each request type will have its own accordion component, right? should this be made in this PR or created in another PR when delivery request is added in?

Yeah, that's right, it should go like this on accordion opened

```
commands 
- command 1
--- Info A1
--- Info A2
- command 2
--- Info B1
--- Info B2
```

And like this on accordion closed

```
commands 
- command 1
- command 2
```

I'll need that for the delivery request PR. I can also make those changes on the Delivery Request PR, but however you want. Let me know.

Reply:
Lets go with your suggestion then! And you can make the changes to the Delivery Request PR as well. Will get this done by today